### ISSUE TITLE
Pretrained Model

### ISSUE THREAD

**Issue by M-SunRise** [xaCheng1996/VFD#3]
Could you provide the pretrained model on Voxceleb2?

**Comment by xaCheng1996** [comment]
We have scheduled to give the pre-trained model and a sample test set (with about 1000 frames). Please wait for a few days since we would rearrange the code first :)

**Comment by xaCheng1996** [comment]
Hi,
The checkpoints are released.

**Comment by M-SunRise** [comment]
I run the code as you mentioned, but I got a wrong result.

The number of test images dir = 99
100%|████████████████████████████████████████████████████████████████████████████████████████████| 99/99 [00:08<00:00, 11.25it/s]
The auc is 0.000

Dataset: FakeAVCeleb 
(download from the google cloud which you offered, as well pretrained models )
Is there any wrong with it?

**Comment by xaCheng1996** [comment]
We are really sorry; this is a minus missing in auc() function. You just need to add a '-' for 'ind' in lines 14 and 17 of test.py:

label_all.append(-ind)
target_all.append(-ind)

This error is because the voice-face distance(score) of real videos is lower than that of fake, and this is considered the wrong classification in auc(). The provided data is relatively simple and the auc for our pre-trained model should be 1.000.

We have fixed this error in the repository. This is a mistake caused by carelessness, we are very sorry.

In addition, we give a histogram demonstration function in test.py (lines 66-87), which should show that real and fake videos are split.

**Comment by M-SunRise** [comment]
Sorry to bother you again.
1. I ran the code on FakeAVCeleb [1.2,](https://github.com/DASH-Lab/FakeAVCeleb) also got the AUC=1.0. I only use RealVideo_RealAudio, and FakeVideo_RealAudio. So in your paper, you also considered the FakeAudio? 

2. Another questiuon, when I run the test_DF.py.
`The number of test images dir = 817` is equal to the true images number. So in the test stage, the fake samples and the true samples are 817? 

3. In the paper, 
> we manuallyfiltered 1,700 real videos with corresponding 5,810 fake videos as the testing set. 
Could you provide the id of these selected videos for us to evalute in the same case?

**Comment by xaCheng1996** [comment]
Hi，
1. We applied the test set of FakeAVCeleb directly without special filtering, so there should be some faked audio.

2. Given that fake videos are far more than real videos, and that we need both real and fake videos when training, we load 1 real video + n fake videos in one batch (in the DFDC_dataset.py, there is 'scale = len(video_fake)/len(video_real)', i.e., scale fake videos are loaded), thus the number of fake samples should be several times the real one. 
`The number of test images dir = 817` is just a rough hint of the size of the test set.

3. Unfortunately, the part of this dataset has been corrupted due to the graduation of one of our co-authors, and we are trying to repair them. 

However, you can actually collect your own test set with DFDC, which should be a fair comparison.

**Comment by M-SunRise** [comment]
HI, I ran the code on FakeAVCeleb again, also included the fake audio. I got the auc=1, that is same as the samples you offered. Then I move some samples which labels are fake to the "real" folder, and tested, got the auc=1, again. I think it must be some wrong with it.
